src/anchoredinterval.jl

```diff
@@ -60,6 +60,17 @@ See also: [`Interval`](@ref), [`HE`](@ref), [`HB`](@ref)
 """
 struct AnchoredInterval{P, T, L <: Bound, R <: Bound} <: AbstractInterval{T,L,R}
     anchor::T
+
+    function AnchoredInterval{P,T,L,R}(anchor::T) where {P, T, L <: Bound, R <: Bound}
+        if P < zero(P) && R === Unbounded || P > zero(P) && L === Unbounded
```

Comment:
Should we really include support for unbounded AnchoredIntervals? Do you expect people to pass infinite periods? I think it makes sense to restrict AnchoredIntervals to finite `P`.

Reply:
A very valid question. Just because we can support it doesn't mean we should. I'll give that some thought

Reply:
One argument for keeping this is that removing `Unbounded` support doesn't stop people from using infinite periods. We'd have to add checks to disable that as well

Reply:
Thinking about this more I think support for unbounded and infinity should be consistent. If we do want to remove unbounded support entirely from `AnchoredInterval` we should also remove support for infinity.

My rational in why these go together is that I was planning on defining equality between infinite and unbounded intervals such that `[∞,x] == (,x]`. Not having unbounded support but having infinity support seems wrong to me considering this.

Reply:
> Thinking about this more I think support for unbounded and infinity should be consistent.

Yeah I do agree.

Okay, I can imagine wanting this rather than just using Interval I guess. But it's hard to imagine why you'd want an AnchoredInterval with infinity as part of the type.

Reply:
Okay, I'm okay with keeping this functionality. But a `MethodError` hurts readability for the error, since the user cannot see from the type signature why this is wrong. I think this should be a different type of error. 

One way to keep it a MethodError would be to add yet another type parameter that's just the sign of the period, then all the relevant information would be available for dispatch.

Reply:
I'll make it an argument error. Not entirely accurate but better

Reply:
More thoughts on this: https://github.com/invenia/Intervals.jl/pull/104#issuecomment-648245619